src/blockchain/utils.ts

```diff
@@ -19,6 +19,10 @@ function getBlockTracker (keyPrefix?: string): BlockTracker {
   return new BlockTracker(confStore)
 }
 
+export async function getBlockDate (eth: Eth, blockNumber: number): Promise<Date> {
+  return new Date(((await eth.getBlock(blockNumber)).timestamp as number) * 1000)
```

Comment:
Maybe it would be clearer to use `Number.parseInt`? Could this typecasting be issue?

Reply:
This should be already a number based on web3js docs. 

> timestamp - Number: The unix timestamp for when the block was collated.

Not sure why I did the TS typecast.